AceComm-3.0/ChatThrottleLib.lua

```diff
@@ -353,6 +423,15 @@ function ChatThrottleLib.OnUpdate(this,delay)
 		return -- argh. some bastard is spewing stuff past the lib. just bail early to save cpu.
 	end
 
+	-- Integrate blocked queues back into their rings periodically.
+	if self.BlockedQueuesDelay >= 1 then
```

Comment:
Options are either lower the timeout or remove it entirely.

As I'd noted in the PR - because blocked pipes are appended to the _end_ of the priority, we have an effect where unblocked pipes implicitly have priority over blocked ones.

This means if the timer were removed and there's other traffic in a priority bracket, it's reasonably likely that we won't try re-sending data from the blocked queues too quickly anyway, as first the round-robin despool logic is going to want to visit the unblocked queues toward the start of the ring before getting to the blocked ones.

As such, the only circumstances at which we might unnecessarily try to re-send data too quickly is if we find ourselves in the situation where a priority bracket has very few unblocked pipes.

Also worth noting that the despooling itself occurs once every 0.08s. We could bump that to about 0.2s, which means that if the blocked queue timer were removed that we're only going to potentially re-send on blocked queues at most 5 times before the prefix is guaranteed to unthrottle itself. I suspect increasing the latency of the despool wouldn't really be too noticeable given that addon comms themselves are going to be subject to variable transmission latency anyway.

Reply:
Blocked queues delay lowered to 0.35s in the squashed version.

Rationale for this figure is that;

1. We want to allow priorities to sit in a blocked state and relinquish their bandwidth for use by other prios that have actively-sendable data. If we removed the timer, this wouldn't be possible.
2. The original 1s value is quite a long time to wait between retries and in some cases may add a lot of artificial delay in scenarios where a prefix peaks over its allocated message count but and then regains a message slot on the next frame or so.
3. Too low values give increase CPU load. Once blocked queues are integrated back into the prio ring, the OnUpdate loop potentially has to walk all of them again in cases where the prio _almost exclusively_ queues that still ultimately end up in the blocked state. The chosen 0.35s figure should mitigate this, as we expect that within 3 attempts a prefix will unblock itself.